# Correlation

## Hypotheis 1
## Null Hypothesis: There is no significant  correlation between Attrition and Monthly Income.
## Alternanative Hypothesis: There is significant correlation between  Attrition and Monthly Income. 

In [3]:
import pandas as pd



In [4]:

dataset1 = pd.read_csv('general_data_set.csv')


In [5]:
dataset1

Age Attrition     BusinessTravel              Department  \
0      51        No      Travel_Rarely                   Sales   
1      31       Yes  Travel_Frequently  Research & Development   
2      32        No  Travel_Frequently  Research & Development   
3      38        No         Non-Travel  Research & Development   
4      32        No      Travel_Rarely  Research & Development   
...   ...       ...                ...                     ...   
4405   42        No      Travel_Rarely  Research & Development   
4406   29        No      Travel_Rarely  Research & Development   
4407   25        No      Travel_Rarely  Research & Development   
4408   42        No      Travel_Rarely                   Sales   
4409   40        No      Travel_Rarely  Research & Development   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                    6          2  Life Sciences              1           1   
1                   10          1  Life Sciences              1           2   
2                   17          4          Other              1           3   
3                    2          5  Life Sciences              1           4   
4                   10          1        Medical              1           5   
...                ...        ...            ...            ...         ...   
4405                 5          4        Medical              1        4406   
4406                 2          4        Medical              1        4407   
4407                25          2  Life Sciences              1        4408   
4408                18          2        Medical              1        4409   
4409                28          3        Medical              1        4410   

      Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0     Female  ...                 1.0      Y                11              8   
1     Female  ...                 0.0      Y                23              8   
2       Male  ...                 1.0      Y                15              8   
3       Male  ...                 3.0      Y                11              8   
4       Male  ...                 4.0      Y                12              8   
...      ...  ...                 ...    ...               ...            ...   
4405  Female  ...                 3.0      Y                17              8   
4406    Male  ...                 2.0      Y                15              8   
4407    Male  ...                 0.0      Y                20              8   
4408    Male  ...                 0.0      Y                14              8   
4409    Male  ...                 0.0      Y                12              8   

      StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
0                    0               1.0                      6   
1                    1               6.0                      3   
2                    3               5.0                      2   
3                    3              13.0                      5   
4                    2               9.0                      2   
...                ...               ...                    ...   
4405                 1              10.0                      5   
4406                 0              10.0                      2   
4407                 0               5.0                      4   
4408                 1              10.0                      2   
4409                 0               NaN                      6   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0                  1                        0                     0  
1                  5                        1                     4  
2                  5                        0                     3  
3                  8                        7                     5  
4                  6                        0                     4  
...              ...                      ...                   ..

In [10]:
dataset9 = pd.read_excel('general_data_set_attrition_yes1.xlsx',sheet_name = 0)

In [11]:
dataset9

Age_Yes Attrition_Yes BusinessTravel_Yes          Department_Yes  \
0          51            No      Travel_Rarely                   Sales   
1          31           Yes  Travel_Frequently  Research & Development   
2          32            No  Travel_Frequently  Research & Development   
3          38            No         Non-Travel  Research & Development   
4          32            No      Travel_Rarely  Research & Development   
...       ...           ...                ...                     ...   
4405       42            No      Travel_Rarely  Research & Development   
4406       29            No      Travel_Rarely  Research & Development   
4407       25            No      Travel_Rarely  Research & Development   
4408       42            No      Travel_Rarely                   Sales   
4409       40            No      Travel_Rarely  Research & Development   

      DistanceFromHome_Yes  Education_Yes EducationField  EmployeeCount_Yes  \
0                        6              2  Life Sciences                  1   
1                       10              1  Life Sciences                  1   
2                       17              4          Other                  1   
3                        2              5  Life Sciences                  1   
4                       10              1        Medical                  1   
...                    ...            ...            ...                ...   
4405                     5              4        Medical                  1   
4406                     2              4        Medical                  1   
4407                    25              2  Life Sciences                  1   
4408                    18              2        Medical                  1   
4409                    28              3        Medical                  1   

      EmployeeID Gender_Yes  ...  NumCompaniesWorked_Yes Over18_Yes  \
0              1     Female  ...                     1.0          Y   
1              2     Female  ...                     0.0          Y   
2              3       Male  ...                     1.0          Y   
3              4       Male  ...                     3.0          Y   
4              5       Male  ...                     4.0          Y   
...          ...        ...  ...                     ...        ...   
4405        4406     Female  ...                     3.0          Y   
4406        4407       Male  ...                     2.0          Y   
4407        4408       Male  ...                     0.0          Y   
4408        4409       Male  ...                     0.0          Y   
4409        4410       Male  ...                     0.0          Y   

     PercentSalaryHike_Yes  StandardHours_Yes  StockOptionLevel_Yes  \
0                       11                  8                     0   
1                       23                  8                     1   
2                       15                  8                     3   
3                       11                  8                     3   
4                       12                  8                     2   
...                    ...                ...                   ...   
4405                    17                  8                     1   
4406                    15                  8                     0   
4407                    20                  8                     0   
4408                    14                  8                     1   
4409                    12                  8                     0   

     TotalWorkingYears_Yes  TrainingTimesLastYear_Yes  YearsAtCompany_Yes  \
0                      1.0                          6                   1   
1                      6.0                          3                   5   
2                      5.0                          2                   5   
3                     13.0                          5                   8   
4                      9.0                          2                   6   


In [10]:
grouped = dataset1.groupby(dataset1.Attrition) 
dataset2 = grouped.get_group("Yes")


In [11]:
dataset2 

Age Attrition     BusinessTravel              Department  \
1      31       Yes  Travel_Frequently  Research & Development   
6      28       Yes      Travel_Rarely  Research & Development   
13     47       Yes         Non-Travel  Research & Development   
28     44       Yes  Travel_Frequently  Research & Development   
30     26       Yes      Travel_Rarely  Research & Development   
...   ...       ...                ...                     ...   
4381   29       Yes      Travel_Rarely  Research & Development   
4386   33       Yes      Travel_Rarely                   Sales   
4388   33       Yes      Travel_Rarely                   Sales   
4391   32       Yes      Travel_Rarely                   Sales   
4402   37       Yes  Travel_Frequently                   Sales   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
1                   10          1  Life Sciences              1           2   
6                   11          2        Medical              1           7   
13                   1          1        Medical              1          14   
28                   1          2        Medical              1          29   
30                   4          3        Medical              1          31   
...                ...        ...            ...            ...         ...   
4381                 7          1  Life Sciences              1        4382   
4386                11          4      Marketing              1        4387   
4388                 1          3  Life Sciences              1        4389   
4391                23          1  Life Sciences              1        4392   
4402                 2          3      Marketing              1        4403   

      Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
1     Female  ...                 0.0      Y                23              8   
6       Male  ...                 2.0      Y                20              8   
13      Male  ...                 1.0      Y                11              8   
28      Male  ...                 3.0      Y                14              8   
30      Male  ...                 2.0      Y                11              8   
...      ...  ...                 ...    ...               ...            ...   
4381  Female  ...                 1.0      Y                21              8   
4386    Male  ...                 5.0      Y                21              8   
4388    Male  ...                 7.0      Y                11              8   
4391    Male  ...                 0.0      Y                11              8   
4402    Male  ...                 6.0      Y                11              8   

      StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
1                    1               6.0                      3   
6                    1               5.0                      2   
13                   2              10.0                      4   
28                   1              19.0                      2   
30                   0               5.0                      5   
...                ...               ...                    ...   
4381                 0               4.0                      2   
4386                 0               8.0                      2   
4388                 0              13.0                      2   
4391                 0               4.0                      2   
4402                 1              17.0                      2   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
1                  5                        1                     4  
6                  0                        0                     0  
13                10                        9                     9  
28                 1                        0                     0  
30                 3                        0                     2  
...              ...                      ...                   ..

In [12]:
grouped = dataset1.groupby(dataset1.Attrition)
dataset3 = grouped.get_group("No")

In [13]:
dataset3

Age Attrition     BusinessTravel              Department  \
0      51        No      Travel_Rarely                   Sales   
2      32        No  Travel_Frequently  Research & Development   
3      38        No         Non-Travel  Research & Development   
4      32        No      Travel_Rarely  Research & Development   
5      46        No      Travel_Rarely  Research & Development   
...   ...       ...                ...                     ...   
4405   42        No      Travel_Rarely  Research & Development   
4406   29        No      Travel_Rarely  Research & Development   
4407   25        No      Travel_Rarely  Research & Development   
4408   42        No      Travel_Rarely                   Sales   
4409   40        No      Travel_Rarely  Research & Development   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                    6          2  Life Sciences              1           1   
2                   17          4          Other              1           3   
3                    2          5  Life Sciences              1           4   
4                   10          1        Medical              1           5   
5                    8          3  Life Sciences              1           6   
...                ...        ...            ...            ...         ...   
4405                 5          4        Medical              1        4406   
4406                 2          4        Medical              1        4407   
4407                25          2  Life Sciences              1        4408   
4408                18          2        Medical              1        4409   
4409                28          3        Medical              1        4410   

      Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0     Female  ...                 1.0      Y                11              8   
2       Male  ...                 1.0      Y                15              8   
3       Male  ...                 3.0      Y                11              8   
4       Male  ...                 4.0      Y                12              8   
5     Female  ...                 3.0      Y                13              8   
...      ...  ...                 ...    ...               ...            ...   
4405  Female  ...                 3.0      Y                17              8   
4406    Male  ...                 2.0      Y                15              8   
4407    Male  ...                 0.0      Y                20              8   
4408    Male  ...                 0.0      Y                14              8   
4409    Male  ...                 0.0      Y                12              8   

      StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
0                    0               1.0                      6   
2                    3               5.0                      2   
3                    3              13.0                      5   
4                    2               9.0                      2   
5                    0              28.0                      5   
...                ...               ...                    ...   
4405                 1              10.0                      5   
4406                 0              10.0                      2   
4407                 0               5.0                      4   
4408                 1              10.0                      2   
4409                 0               NaN                      6   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0                  1                        0                     0  
2                  5                        0                     3  
3                  8                        7                     5  
4                  6                        0                     4  
5                  7                        7                     7  
...              ...                      ...                   ..

In [23]:
from sklearn import preprocessing

Label_encoder = preprocessing.LabelBinarizer() 
dataset1["Attrition"] = Label_encoder.fit_transform (dataset1["Attrition"])

from scipy.stats import pearsonr

stats,p = pearsonr(dataset1.Attrition, dataset1.MonthlyIncome)
print(stats,p)


-0.031176281698115017 0.03842748490600132


## Inference

__since p = 0.038427<0.05 accept Alternative hypothesis__
__i.e "There is significant correlation between  Attrition and Monthly Income".__

# Hypotheis 2
## Null Hypothesis: There is no significant  correlation between Attrition and EducationField.
## Alternanative Hypothesis: There is significant correlation between  Attrition and EducationField.

In [24]:
from sklearn import preprocessing

Label_encoder = preprocessing.LabelBinarizer()
dataset1["Attrition"] = Label_encoder.fit_transform(dataset1["Attrition"])
dataset1["EducationField"] = Label_encoder.fit_transform(dataset1["EducationField"])

from scipy.stats import pearsonr

stats,p = pearsonr(dataset1.Attrition, dataset1.EducationField)
print(stats,p)

0.0915731712401226 1.1133098147978756e-09


__since p = 1.113309814e-09<0.05 i.e "There is significant correlation between  Attrition and EducationField".__

# Hypotheis 3
## Null Hypothesis: There is no significant  correlation between Attrition and Age.
## Alternanative Hypothesis: There is significant correlation between  Attrition and Age


In [26]:
from sklearn import preprocessing

Label_encoder = preprocessing.LabelBinarizer()
dataset1["Attrition"] = Label_encoder.fit_transform(dataset1["Attrition"])

from scipy.stats import pearsonr

stats, p = pearsonr(dataset1.Attrition,dataset1.Age)
print("statistical value = ",stats )
print("probability value = ",p)

statistical value =  -0.15920500686577943
probability value =  1.996801615887666e-26


__since p = 1.113309814e-09<0.05 i.e "There is significant correlation between  Attrition and EducationField".__

# Hypotheis 3
## Null Hypothesis: There is no significant  correlation between Attrition and PercentSalaryHike.
## Alternanative Hypothesis: There is significant correlation between  Attrition and PercentSalaryHike


In [29]:
from sklearn import preprocessing

Label_encoder = preprocessing.LabelBinarizer()
dataset1["Attrition"] = Label_encoder.fit_transform(dataset1["Attrition"])

from scipy.stats import pearsonr
stats, p = pearsonr(dataset1.Attrition,dataset1.PercentSalaryHike)
print("statistical value = ",stats )
print("probability value = ",p)

statistical value =  0.032532594891053514
probability value =  0.030743386433355353


__since p = 0.03259489105315<0.05 i.e "There is significant correlation between  Attrition and PercentSalaryHike".__

__Therefor Alternative Hypothesis is accepted__

# Hypotheis 4
## Null Hypothesis: There is no significant  correlation between Attrition and TotalWorkingYears
.
## Alternanative Hypothesis: There is significant correlation between  Attrition and TotalWorkingYears

In [5]:
from sklearn import preprocessing
dataset1.drop_duplicates()

Label_encoder = preprocessing.LabelBinarizer()
dataset1["Attrition"] = Label_encoder.fit_transform(dataset1["Attrition"])

from scipy.stats import pearsonr
stats,p = pearsonr(dataset1.Attrition, dataset1.TotalWorkingYears)
print("statistical value = ",stats)
print("probability value = ",p)

ValueError: array must not contain infs or NaNs

# Hypotheis 5
## Null Hypothesis: There is no significant  correlation between Attrition and Gender
.
## Alternanative Hypothesis: There is significant correlation between  Attrition and Gender

In [6]:
from sklearn import preprocessing

Label_encoder = preprocessing.LabelBinarizer()
dataset1["Attrition"] = Label_encoder.fit_transform(dataset1["Attrition"])
dataset1["Gender"] = Label_encoder.fit_transform(dataset1["Gender"])

from scipy.stats import pearsonr
stats, p = pearsonr(dataset1.Attrition,dataset1.Gender)
print("statistical value = ",stats )
print("probability value = ",p)

statistical value =  0.018125078877010273
probability value =  0.22881970951790567


__since p = 0.22881970951790567>0.05  accept Null hypothesis__
__i.e "There is no significant correlation between  Attrition and Gender".__

# Hypotheis 5
## Null Hypothesis: There is no significant  correlation between Attrition and NumCompaniesWorked


## Alternanative Hypothesis: There is significant correlation between  Attrition and NumCompaniesWorked


In [7]:
from sklearn import preprocessing

Label_encoder = preprocessing.LabelBinarizer()
dataset1["Attrition"] = Label_encoder.fit_transform(dataset1["Attrition"])
dataset1["NumCompaniesWorked"] = Label_encoder.fit_transform(dataset1["NumCompaniesWorked"])


from scipy.stats import pearsonr
stats, p = pearsonr(dataset1.Attrition,dataset1.NumCompaniesWorked)
print("statistical value = ",stats )
print("probability value = ",p)

ValueError: array must not contain infs or NaNs